pages/treeherder.py

```diff
@@ -59,10 +52,79 @@ def open_perfherder_page(self):
 
-    def open_single_resultset(self):
-        Wait(self.selenium, self.timeout).until(
-            EC.visibility_of_element_located(self._first_resultset_datestamp_locator))
-        self.selenium.find_element(*self._first_resultset_datestamp_locator).click()
+    def select_next_job(self):
+        el = self.selenium.find_element(*self._result_set_locator)
+        Wait(self.selenium, self.timeout).until(EC.visibility_of(el))
+        el.send_keys(Keys.ARROW_RIGHT)
+        Wait(self.selenium, self.timeout).until(lambda s: self.result.job_result_status)
+        return self.selenium.find_element(*self._selected_job_title_locator).get_attribute('title')
+
+    class JobDetails(PageRegion):
+
+        _job_details_actionbar_locator = (By.ID, 'job-details-actionbar')
+        _job_details_pin_job_locator = (By.CSS_SELECTOR, '#job-details-actionbar .nav .nav li:nth-child(1) a')
+        _logviewer_button_locator = (By.ID, 'logviewer-btn')
+        _result_status_locator = (By.ID, 'job-details-panel')
+
+        def wait_for_page_to_load(self):
+            body = self.find_element(By.TAG_NAME, 'body')
+            self.wait.until(lambda s: 'loaded' in body.get_attribute('JobDetails'))
+
+        def open_logviewer(self):
+            self.wait_for_page_to_load
+            self.selenium.find_element(*self._result_status_locator).send_keys('l')
+            return LogviewerPage(self.base_url, self.selenium)
+
+        def pin_job(self):
+            self.wait_for_page_to_load
+            self.selenium.find_element(*self._job_details_pin_job_locator).click()
+
+    class Pinboard(PageRegion):
+
+        _root_locator = (By.ID, 'pinboard-panel')
+
+        _clear_all_menu_locator = (By.CSS_SELECTOR, '#pinboard-controls .dropdown-menu li:nth-child(4)')
+        _open_save_menu_locator = (By.CSS_SELECTOR, '#pinboard-controls span.dropdown-toggle > span:nth-child(1)')
+        _pinboard_jobs_locator = (By.CSS_SELECTOR, '#pinned-job-list .pinned-job')
+        _pinboard_remove_job_locator = (By.CSS_SELECTOR, '#pinned-job-list .pinned-job-close-btn')
+        _pinned_job_title_locator = (By.CSS_SELECTOR, '.pinned-job.selected-job')
+
+        @property
+        def pinned_job_title(self):
+            return self.selenium.find_element(*self._pinned_job_title_locator).get_attribute('title')
+
+        @property
+        def pins(self):
+            return len(self.selenium.find_elements(*self._pinboard_jobs_locator))
+
+        @property
+        def is_pinboard_open(self):
+            return self.is_element_visible(self._root_locator)
+
+        def clear_pinboard(self):
+            self.selenium.find_element(*self._open_save_menu_locator).click()
+            self.selenium.find_element(*self._clear_all_menu_locator).click()
+
+    class Result(PageRegion):
+
+        _root_locator = (By.ID, 'info-panel-content')
+
+        _job_result_status_locator = (By.CSS_SELECTOR, '#result-status-pane > div:nth-child(1) > span:nth-child(2)')
```

Comment:
The intention was for JobDetails to be the specific job results pane and the Result region to be for jobs listed within a result-set. After re-jiggering I realized I created the Result region for no real reason- so this should be cleared up with my patch.